ezgui/src/runner.rs

```diff
@@ -315,34 +317,45 @@ pub fn run<G: 'static + GUI, F: FnOnce(&mut EventCtx) -> G>(settings: Settings,
                 winit::event_loop::ControlFlow::WaitUntil(Instant::now() + UPDATE_FREQUENCY);
         }
 
-        let (mode, input_used) = state.event(ev, &prerender);
+        let (updates, input_used) = state.event(ev, &prerender);
+
         if input_used {
             prerender.request_redraw();
         }
 
-        match mode {
-            EventLoopMode::InputOnly => {
-                running = false;
-                *control_flow = winit::event_loop::ControlFlow::Wait;
-            }
-            EventLoopMode::Animation => {
-                // If we just unpaused, then don't act as if lots of time has passed.
-                if !running {
-                    last_update = Instant::now();
-                    *control_flow = winit::event_loop::ControlFlow::WaitUntil(
-                        Instant::now() + UPDATE_FREQUENCY,
-                    );
-                }
 
-                running = true;
-            }
-            EventLoopMode::ScreenCaptureEverything {
-                dir,
-                zoom,
-                max_x,
-                max_y,
-            } => {
-                screenshot_everything(&mut state, &dir, &prerender, zoom, max_x, max_y);
+        for update in updates {
+            match update {
+                UpdateType::InputOnly => {
+                    running = false;
+                    *control_flow = winit::event_loop::ControlFlow::Wait;
+                },
+                UpdateType::Game => {
+                    // If we just unpaused, then don't act as if lots of time has passed.
+                    if !running {
+                        last_update = Instant::now();
+                        *control_flow = winit::event_loop::ControlFlow::WaitUntil(
+                            Instant::now() + UPDATE_FREQUENCY,
+                        );
+                    }
+
+                    running = true;
+                },
+                UpdateType::Pan{
+                    delta_x,
+                    delta_y
+                } => {
+                    state.canvas.move_camera(delta_x,
```

Comment:
So this doesn't touch `running` at all... huh. To deal with the issue of auto-panning changing speed based on wiggling the mouse: Maybe we should only trigger UpdateType::Pan when `input.nonblocking_is_update_event().is_some()`? But the game and pan updates are hard to distinguish.

I wonder if there's a way to queue multiple user-defined events in the future. When the sim is running from sandbox/speed, we request a game update event in X seconds. While the mouse is hovering along the border, we request a pan update event in Y seconds. That's the API that makes sense in my head. But I'm not sure how to wrangle the ControlFlow stuff to do that. It might not be possible; we might need some kind of extra state that tracks what updates have been requested, does WaitUntil the minimum amount of time, then fires the appropriate event(s)

Reply:
So, it seems using `input.nonblocking_is_update_event().is_some()` alleviates the issue with changing speed based on mouse wiggling. Is there still value in exploring future queuing then?

Reply:
I guess not. The only issue now is that auto-pan speed seems to be a little different when the sim is paused or not. Not entirely sure what's happening there, but the functionality now feels good. The main problem was the mouse wiggling causing speed up, which is fixed now

Reply:
Okay, if we merge this in I'll make an issue to track the bug with regard to different speeds in paused vs running